prometheus/counter_test.go

```diff
@@ -273,3 +273,52 @@ func TestCounterExemplar(t *testing.T) {
 		t.Error("adding exemplar with oversized labels succeeded")
 	}
 }
+
+func TestCounterCreatedTimestamp(t *testing.T) {
+	now := time.Now()
+
+	counter := NewCounter(CounterOpts{
+		Name: "test",
+		Help: "test help",
+	}).(*counter)
+	counter.now = func() time.Time { return now }
```

Comment:
I don't get this test. We should test NewCounter giving us correct CT. Instead we manually override it, which won't test real scenario. Does it make sense? 🤔

Reply:
Exactly, that's why I pinged you in slack 😅


`NewCounter` doesn't have access to counter.now, so I can't crest a counter with a mocked now function. I'm struggling to find a good way to test NewCounter an NewCounterVec for that reason

Reply:
Yea, we have to do bit of improvisation here. One idea is to have package-level, private variable we will replace in tests and it's always time.Now for production.

Reply:
var timeNow func() time.Time = time.Now

Reply:
```
counter := NewCounter(CounterOpts{
		Name: "test",
		Help: "test help",

                 timeNow: func() time.Time { return now }
	})
```